packages/common/http/src/fetch.ts

```diff
@@ -108,29 +109,32 @@ export class FetchBackend implements HttpBackend {
       let decoder: TextDecoder;
       let partialText: string|undefined;
 
-      while (true) {
-        const {done, value} = await reader.read();
-
-        if (done) {
-          break;
-        }
-
-        chunks.push(value);
-        receivedLength += value.length;
-
-        if (request.reportProgress) {
-          partialText = request.responseType === 'text' ?
-              (partialText ?? '') + (decoder ??= new TextDecoder).decode(value, {stream: true}) :
-              undefined;
-
-          observer.next({
-            type: HttpEventType.DownloadProgress,
-            total: contentLength ? +contentLength : undefined,
-            loaded: receivedLength,
-            partialText,
-          } as HttpDownloadProgressEvent);
+      const reqZone = Zone.current;
+      await this.ngZone.runOutsideAngular(async () => {
```

Comment:
Minor proposal: let's add a comment in the code to describe why the `runOutsideAngular` is needed. Something like this:

```suggestion
      // Perform response processing outside of Angular zone to
      // ensure no excessive change detection runs are executed. 
      await this.ngZone.runOutsideAngular(async () => {
```

It'd be also great to add to that comment what exactly is triggering change detection (so it's easier to support this code in the future).